src/hooks/use-scroll.ts

```diff
@@ -0,0 +1,54 @@
+import { RefObject, useState } from 'react'
+import { AxisScrollInfo, scroll, ScrollOptions } from '@motionone/dom'
+import { useIsomorphicLayoutEffect } from 'hooks/use-isomorphic-layout-effect'
+
+interface UseScrollOptions extends Omit<ScrollOptions, 'container' | 'target'> {
+	container?: RefObject<HTMLElement>
+	target?: RefObject<HTMLElement>
+}
+
+interface ScrollValues {
+	scrollX: number
+	scrollXProgress: number
+	scrollY: number
+	scrollYProgress: number
+}
+
+/**
+ * Track scroll metrics.
+ * Uses `@motionone/dom` under the hood.
+ *
+ * https://motion.dev/dom/scroll
+ */
+export default function useScroll({
+	container,
+	target,
+	...options
+}: UseScrollOptions = {}) {
+	const [scrollValues, setScrollValues] = useState<ScrollValues>({
+		scrollX: 0,
+		scrollXProgress: 0,
+		scrollY: 0,
+		scrollYProgress: 0,
+	})
+
+	useIsomorphicLayoutEffect(() => {
+		return scroll(
```

Comment:
Taking a very similar approach to what `framer-motion` does internally, using `@motionone`. But this way, we don't need to require the use of `m.div` to use the return values of `useScroll`.